openmc/deplete/helpers.py

```diff
@@ -92,6 +107,153 @@ def get_material_rates(self, mat_id, nuc_index, react_index):
         return self._results_cache
 
 
+class FluxCollapseHelper(ReactionRateHelper):
+    """Class that generating one-group reaction rates using multigroup flux
+
+    This class generates a multigroup flux tally that is used afterward to
+    calculate a one-group reaction rate by collapsing it with continuous-energy
+    cross section data. Additionally, select nuclides/reactions can be treated
+    with a direct reaction rate tally when using a multigroup flux spectrum
+    would not be sufficiently accurate. This is often the case for (n,gamma) and
+    fission reactions.
+
+    .. versionadded:: 0.12.1
+
+    Parameters
+    ----------
+    n_nucs : int
+        Number of burnable nuclides tracked by :class:`openmc.deplete.Operator`
+    n_react : int
+        Number of reactions tracked by :class:`openmc.deplete.Operator`
+    energies : iterable of float
+        Energy group boundaries for flux spectrum in [eV]
+    reactions : iterable of str
+        Reactions for which rates should be directly tallied
+    nuclides : iterable of str
+        Nuclides for which some reaction rates should be directly tallied. If
+        None, then ``reactions`` will be used for all nuclides.
+
+    Attributes
+    ----------
+    nuclides : list of str
+        All nuclides with desired reaction rates.
+
+    """
+    def __init__(self, n_nucs, n_reacts, energies, reactions=None, nuclides=None):
+        super().__init__(n_nucs, n_reacts)
+        self._energies = asarray(energies)
+        self._reactions_direct = list(reactions) if reactions is not None else []
+        self._nuclides_direct = list(nuclides) if nuclides is not None else None
+
+    @ReactionRateHelper.nuclides.setter
+    def nuclides(self, nuclides):
+        ReactionRateHelper.nuclides.fset(self, nuclides)
+        if self._reactions_direct and self._nuclides_direct is None:
+            self._rate_tally.nuclides = nuclides
+
+    def generate_tallies(self, materials, scores):
+        """Produce multigroup flux spectrum tally
+
+        Uses the :mod:`openmc.lib` module to generate a multigroup flux tally
+        for each burnable material.
+
+        Parameters
+        ----------
+        materials : iterable of :class:`openmc.Material`
+            Burnable materials in the problem. Used to construct a
+            :class:`openmc.MaterialFilter`
+        scores : iterable of str
+            Reaction identifiers, e.g. ``"(n, fission)"``, ``"(n, gamma)"``,
+            needed for the reaction rate tally.
+        """
+        self._materials = materials
+
+        # Convert reactions to MT values (needed when collapsing)
+        mt_values = {v: k for k, v in REACTION_NAME.items()}
```

Comment:
Would this be a useful object to provide as part of the `openmc.data.reaction` module?

Reply:
I haven't really found other uses for this. In this particular case, it was more convenient to pass an MT value down to the C++  level rather than passing a `char*` and then having to determine the MT value on the C++ side (if you look at `Nuclide::collapse_rate`, you see that we use the MT value as an index in `reaction_index_`).